presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -61,14 +62,14 @@
     @Override
     public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
     {
-        return rewriteWith(new InPredicateRewriter(idAllocator), plan, null);
+        return rewriteWith(new InPredicateRewriter(idAllocator), plan, session);
     }
 
     /**
      * Each ApplyNode which contains InPredicate is replaced by semi join node.
      */
     private static class InPredicateRewriter
-            extends SimplePlanRewriter<Void>
+            extends SimplePlanRewriter<Session>
```

Comment:
You don't have to pass session via context at all. Yo can just make it field in `InPredicateRewriter`